xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1097,3 +1126,190 @@ void DX::DeviceResources::Trim() const
+        }
+      }
+      else
+      {
+        CLog::LogF(LOGERROR, "DXGI GetDesc1 failed");
+      }
+    }
+  }
+
+  return hdr10;
+}
+
+void DX::DeviceResources::SetHdrMetaData(DXGI_HDR_METADATA_HDR10& hdr10) const
+{
+  ComPtr<IDXGISwapChain4> swapChain4;
+
+  if (m_swapChain == nullptr)
+    return;
+
+  if (SUCCEEDED(m_swapChain.As(&swapChain4)))
+  {
+    if (SUCCEEDED(swapChain4->SetHDRMetaData(DXGI_HDR_METADATA_TYPE_HDR10, sizeof(hdr10), &hdr10)))
+    {
+      constexpr double FACTOR_1 = 50000.0;
+      constexpr double FACTOR_2 = 10000.0;
+      const double RP_0 = static_cast<double>(hdr10.RedPrimary[0]) / FACTOR_1;
+      const double RP_1 = static_cast<double>(hdr10.RedPrimary[1]) / FACTOR_1;
+      const double GP_0 = static_cast<double>(hdr10.GreenPrimary[0]) / FACTOR_1;
+      const double GP_1 = static_cast<double>(hdr10.GreenPrimary[1]) / FACTOR_1;
+      const double BP_0 = static_cast<double>(hdr10.BluePrimary[0]) / FACTOR_1;
+      const double BP_1 = static_cast<double>(hdr10.BluePrimary[1]) / FACTOR_1;
+      const double WP_0 = static_cast<double>(hdr10.WhitePoint[0]) / FACTOR_1;
+      const double WP_1 = static_cast<double>(hdr10.WhitePoint[1]) / FACTOR_1;
+      const double Max_ML = static_cast<double>(hdr10.MaxMasteringLuminance) / FACTOR_2;
+      const double min_ML = static_cast<double>(hdr10.MinMasteringLuminance) / FACTOR_2;
+
+      CLog::LogF(LOGNOTICE,
+                 "Stream HDR metadata => RP {0:0.3f} {1:0.3f} | GP {2:0.3f} {3:0.3f} | BP "
+                 "{4:0.3f} {5:0.3f} | WP {6:0.3f} "
+                 "{7:0.3f} | MAX ML {8:0.0f} "
+                 "| min ML {9:0.3f} | Max CLL {10:d} | Max FALL {11:d}",
+                 RP_0, RP_1, GP_0, GP_1, BP_0, BP_1, WP_0, WP_1, Max_ML, min_ML,
+                 hdr10.MaxContentLightLevel, hdr10.MaxFrameAverageLightLevel);
+    }
+    else
+    {
+      CLog::LogF(LOGERROR, "DXGI SetHDRMetaData failed");
+    }
+  }
+}
+
+void DX::DeviceResources::SetHdrColorSpace(const DXGI_COLOR_SPACE_TYPE colorSpace) const
+{
+  ComPtr<IDXGISwapChain3> swapChain3;
+
+  if (m_swapChain == nullptr)
+    return;
+
+  if (SUCCEEDED(m_swapChain.As(&swapChain3)))
+  {
+    DXGI_COLOR_SPACE_TYPE cs = colorSpace;
+    if (DX::Windowing()->UseLimitedColor())
+    {
+      switch (cs)
+      {
+        case DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709:
+          cs = DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P709;
+          break;
+        case DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020:
+          cs = DXGI_COLOR_SPACE_RGB_STUDIO_G2084_NONE_P2020;
+          break;
+        case DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P2020:
+          cs = DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P2020;
+          break;
+      }
+    }
+    if (SUCCEEDED(swapChain3->SetColorSpace1(cs)))
+    {
+      CLog::LogF(LOGDEBUG, "DXGI SetColorSpace1 success");
+    }
+    else
+    {
+      CLog::LogF(LOGERROR, "DXGI SetColorSpace1 failed");
+    }
+  }
+}
+
+bool DX::DeviceResources::IsDisplayHDREnabled() const
```

Comment:
> You can skip the util class, and just put it in the windowing method,

FALSE because => A static method of a class cannot be overridden by derived class.

Also `bool CWinSystemAndroid::IsHDRDisplay() ` returns bool

In Windows needed 3 states:

0 => No display HDR capable;
1 => Display HDR capable but not enabled (because Windows switch = OFF)
2 => Display HDR capable and enabled.

But.....................

`bool DX::DeviceResources::IsDisplayHDREnabled()`

is a alternate **fast** version of `int CWIN32Util::GetHDRDisplayStatus()`

to use inside `DX::DeviceResources::ResizeBuffers()`  this is not needed in Windowing because is a local method only used in DX::DeviceResources.

Also:

Windowing() methods:   "override" cannot be used at Application Log start (Application.cpp)  
because `CServiceBroker::GetWinSystem()->`   returns nullptr    because it is not yet initialized.

Due this `GetHDRDisplayStatus()`  needs to be in `CWIN32Util` as a static method.

Reply:
Just for info, I personally would refactor the method as it's actually doing 2 checks

1. Is the display HDR capable
2. Is windows toggle on/off

Again, dealer's choice, I have no skin in the game when it comes to windows, take anything I say as just suggestions regarding this PR

Reply:
Functionally it goes perfect. I am using it (unlike all of you).